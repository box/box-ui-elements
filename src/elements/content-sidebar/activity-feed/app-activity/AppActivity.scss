@import '../../../common/variables';

.bcs-AppActivity {
    width: 100%;

    &.bcs-is-pending {
        opacity: .4;
    }
}

.bcs-AppActivity-confirm {
    z-index: $overlay-z-index;
}

.bcs-AppActivity-content {
    color: $bdl-gray-80;
}

.bcs-AppActivity-icon {
    width: 32px;
    height: 32px;
}

.bcs-AppActivity-link {
    &.link {
        color: $blue;
    }
}

.bcs-AppActivity-headline {
    color: $bdl-gray;
    font-weight: bold;
}
