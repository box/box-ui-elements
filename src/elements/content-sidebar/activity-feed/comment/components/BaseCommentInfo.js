// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import { ACTIVITY_TARGETS } from '../../../../common/interactionTargets';
import { PLACEHOLDER_USER } from '../../../../../constants';
import ActivityTimestamp from '../../common/activity-timestamp';
import Avatar from '../../Avatar';
import IconAnnotation from '../../../../../icons/two-toned/IconAnnotation';
import UserLink from '../../common/user-link';

import type { FeedItemStatus } from '../../../../../common/types/feed';
import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../../common/flowTypes';
import type { User } from '../../../../../common/types/core';

import messages from '../messages';

import './BaseCommentInfo.scss';

export interface BaseCommentInfoProps {
    annotationActivityLink?: React.Element<any> | typeof undefined;
    created_at: string | number;
    created_by: User;
    getAvatarUrl: GetAvatarUrlCallback;
    getUserProfileUrl?: GetProfileUrlCallback | typeof undefined;
    status?: FeedItemStatus | typeof undefined;
}

export const BaseCommentInfo = ({
    annotationActivityLink,
    created_at,
    created_by,
    getAvatarUrl,
    getUserProfileUrl,
}: BaseCommentInfoProps) => {
    const createdByUser = created_by || PLACEHOLDER_USER;
    const createdAtTimestamp = new Date(created_at).getTime();

    return (
        <div className="bcs-BaseCommentInfo">
            <div className="bcs-BaseCommentInfo-avatar">
                <Avatar
                    badgeIcon={
                        annotationActivityLink && (
                            <IconAnnotation
                                title={<FormattedMessage {...messages.inlineCommentAnnotationIconTitle} />}
                            />
                        )
                    }
                    getAvatarUrl={getAvatarUrl}
                    user={createdByUser}
                />
            </div>
            <div className="bcs-BaseCommentInfo-data">
                <div className="bcs-Comment-headline">
                    <UserLink
                        data-resin-target={ACTIVITY_TARGETS.PROFILE}
                        getUserProfileUrl={getUserProfileUrl}
                        id={createdByUser.id}
                        name={createdByUser.name}
                    />
                </div>
                <div className="bcs-BaseCommentInfo-data-timestamp">
                    <ActivityTimestamp date={createdAtTimestamp} />
                    {annotationActivityLink && (
                        <div className="bcs-BaseComment-AnnotationActivityLink">{annotationActivityLink}</div>
                    )}
                </div>
            </div>
        </div>
    );
};
