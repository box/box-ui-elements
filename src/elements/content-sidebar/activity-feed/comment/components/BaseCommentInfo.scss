@import '../../../../common/variables';

.bcs-BaseCommentInfo {
    display: flex;
    align-items: 'bottom';
    justify-content: flex-start;
    padding: 2;
    font-size: $bdl-fontSize--dejaBlue;

    .bdl-Media-figure {
        line-height: 0;
    }
}

.bcs-BaseCommentInfo-avatar {
    display: flex;
    flex-direction: column;
    height: 100%;
}

.bcs-BaseCommentInfo-data {
    display: flex;
    flex-direction: column;
    margin-left: 10px;

    .bcs-ActivityTimestamp {
        color: $bdl-gray-65;
    }

    .bcs-Comment-headline {
        font-weight: bold;
    }
}

.bcs-BaseCommentInfo-data-timestamp {
    display: flex;
    color: $bdl-gray;
}
