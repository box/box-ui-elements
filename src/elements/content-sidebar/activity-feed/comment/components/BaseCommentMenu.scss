.BaseCommentMenu.bdl-Media-menu {
    // to override float:right from .bdl-Media-menu
    float: none;
}
