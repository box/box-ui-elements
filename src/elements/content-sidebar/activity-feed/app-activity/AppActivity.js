/**
 * @flow
 * @file AppActivity component
 */

import * as React from 'react';
import classNames from 'classnames';
import getProp from 'lodash/get';
import noop from 'lodash/noop';
import TetherComponent from 'react-tether';
import { FormattedMessage, injectIntl } from 'react-intl';

import ActivityCard from '../ActivityCard';
import ActivityTimestamp from '../common/activity-timestamp';
import DeleteConfirmation from '../common/delete-confirmation';
import IconTrash from '../../../../icons/general/IconTrash';
import Media from '../../../../components/media';
import messages from './messages';
import { bdlGray80 } from '../../../../styles/variables';
import { Link } from '../../../../components/link';
import { MenuItem } from '../../../../components/menu';
import type { AppItem, ActivityTemplateItem, ActionItemError } from '../../../../common/types/feed';
import type { User, BoxItemPermission } from '../../../../common/types/core';
import './AppActivity.scss';

type Props = {
    activity_template: ActivityTemplateItem,
    app: AppItem,
    created_at: string,
    created_by: User,
    currentUser?: User,
    error?: ActionItemError,
    id: string,
    intl: any,
    isPending?: boolean,
    onDelete: ({ id: string, permissions?: {} }) => void,
    permissions?: BoxItemPermission,
    rendered_text: string,
};

type State = {
    isConfirmingDelete: boolean,
};

function mapActivityNodes(node: HTMLLinkElement): React.Node {
    const { dataset = {}, href = '#', tagName, textContent } = node;

    switch (tagName) {
        case 'A':
            return (
                <Link
                    href={href}
                    data-resin-target={dataset.resinTarget}
                    data-resin-action={dataset.resinAction}
                    key={`app_actvity_link_${href}`}
                    rel="roreferrer noopener"
                    className="bcs-AppActivity-link"
                    target="_blank"
                >
                    {textContent}
                </Link>
            );
        default:
            return textContent;
    }
}

class AppActivity extends React.PureComponent<Props, State> {
    static defaultProps = {
        onDelete: noop,
        permissions: {},
    };

    parser = new DOMParser();

    state: State = {
        isConfirmingDelete: false,
    };

    handleDeleteCancel = (): void => {
        this.setState({ isConfirmingDelete: false });
    };

    handleDeleteClick = () => {
        this.setState({ isConfirmingDelete: true });
    };

    handleDeleteConfirm = (): void => {
        const { id, onDelete, permissions } = this.props;

        onDelete({ id, permissions });
    };

    parseActivity = (): Array<HTMLLinkElement> => {
        const { rendered_text: renderedText } = this.props;
        const doc: Document = this.parser.parseFromString(renderedText, 'text/html');
        if (!doc) {
            return [];
        }

        const childNodes = getProp(doc, 'body.childNodes', []);
        return Array.from(childNodes);
    };

    render() {
        const {
            activity_template: { id: templateId },
            app: { name, icon_url },
            created_at: createdAt,
            created_by: createdBy,
            currentUser,
            error,
            intl,
            isPending,
            permissions,
        } = this.props;

        const canDelete = getProp(permissions, 'can_delete', false) || (currentUser && currentUser.id === createdBy.id);
        const createdAtTimestamp = new Date(createdAt).getTime();
        const isMenuVisible = canDelete && !isPending;
        const { isConfirmingDelete } = this.state;

        return (
            <ActivityCard
                className="bcs-AppActivity"
                data-resin-target="loaded"
                data-resin-feature={`appActivityCard${templateId}`}
            >
                <Media
                    className={classNames({
                        'bcs-is-pending': isPending || error,
                    })}
                >
                    <Media.Figure>
                        <img
                            className="bcs-AppActivity-icon"
                            alt={intl.formatMessage(messages.appActivityAltIcon, { appActivityName: name })}
                            src={icon_url}
                        />
                    </Media.Figure>

                    <Media.Body className="bcs-AppActivity-body">
                        {isMenuVisible && (
                            <TetherComponent
                                attachment="top right"
                                className="bcs-AppActivity-confirm"
                                constraints={[{ to: 'scrollParent', attachment: 'together' }]}
                                targetAttachment="bottom right"
                            >
                                <Media.Menu isDisabled={isConfirmingDelete}>
                                    <MenuItem onClick={this.handleDeleteClick}>
                                        <IconTrash color={bdlGray80} />
                                        <FormattedMessage {...messages.appActivityDeleteMenuItem} />
                                    </MenuItem>
                                </Media.Menu>

                                {isConfirmingDelete && (
                                    <DeleteConfirmation
                                        isOpen={isConfirmingDelete}
                                        message={messages.appActivityDeletePrompt}
                                        onDeleteCancel={this.handleDeleteCancel}
                                        onDeleteConfirm={this.handleDeleteConfirm}
                                    />
                                )}
                            </TetherComponent>
                        )}

                        <figcaption className="bcs-AppActivity-headline">{name}</figcaption>

                        <div>
                            <ActivityTimestamp date={createdAtTimestamp} />
                        </div>

                        {this.parseActivity().map(mapActivityNodes)}
                    </Media.Body>
                </Media>
            </ActivityCard>
        );
    }
}

export default injectIntl(AppActivity);
