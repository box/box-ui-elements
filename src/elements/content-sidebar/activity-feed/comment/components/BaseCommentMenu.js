// @flow
import * as React from 'react';

import { FormattedMessage } from 'react-intl';
import TetherComponent from 'react-tether';

import { ACTIVITY_TARGETS } from '../../../../common/interactionTargets';
import { COMMENT_STATUS_OPEN, COMMENT_STATUS_RESOLVED } from '../../../../../constants';
import { MenuItem } from '../../../../../components/menu';

import Checkmark16 from '../../../../../icon/line/Checkmark16';
import DeleteConfirmation from '../../common/delete-confirmation';
import Media from '../../../../../components/media';
import messages from '../messages';
import Pencil16 from '../../../../../icon/line/Pencil16';
import Trash16 from '../../../../../icon/line/Trash16';
import X16 from '../../../../../icon/fill/X16';

import type { FeedItemStatus } from '../../../../../common/types/feed';

import './BaseCommentMenu.scss';

export interface BaseCommentMenuProps {
    canDelete: boolean;
    canEdit: boolean;
    canResolve: boolean;
    handleDeleteCancel: () => void;
    handleDeleteClick: () => void;
    handleDeleteConfirm: () => void;
    handleEditClick: () => void;
    handleMenuClose: () => void;
    handleStatusUpdate: (selectedStatus: FeedItemStatus) => void;
    isConfirmingDelete: boolean;
    isResolved: boolean;
    onSelect: (isSelected: boolean) => void;
}
export const BaseCommentMenu = ({
    canDelete,
    canEdit,
    canResolve,
    handleDeleteCancel,
    handleDeleteClick,
    handleDeleteConfirm,
    handleEditClick,
    handleMenuClose,
    handleStatusUpdate,
    isConfirmingDelete,
    isResolved,
    onSelect,
}: BaseCommentMenuProps) => {
    return (
        <TetherComponent
            attachment="top right"
            className="bcs-Comment-deleteConfirmationModal"
            constraints={[{ to: 'scrollParent', attachment: 'together' }]}
            targetAttachment="bottom right"
        >
            <Media.Menu
                data-testid="comment-actions-menu"
                dropdownProps={{
                    onMenuOpen: () => onSelect(true),
                    onMenuClose: handleMenuClose,
                }}
                className="BaseCommentMenu"
                isDisabled={isConfirmingDelete}
                menuProps={{
                    'data-resin-component': ACTIVITY_TARGETS.COMMENT_OPTIONS,
                }}
            >
                {canResolve && isResolved && (
                    <MenuItem
                        className="bcs-Comment-unresolveComment"
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="unresolve-comment"
                        onClick={() => handleStatusUpdate(COMMENT_STATUS_OPEN)}
                    >
                        <X16 />
                        <FormattedMessage {...messages.commentUnresolveMenuItem} />
                    </MenuItem>
                )}
                {canResolve && !isResolved && (
                    <MenuItem
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="resolve-comment"
                        onClick={() => handleStatusUpdate(COMMENT_STATUS_RESOLVED)}
                    >
                        <Checkmark16 />
                        <FormattedMessage {...messages.commentResolveMenuItem} />
                    </MenuItem>
                )}
                {canEdit && (
                    <MenuItem
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="edit-comment"
                        onClick={handleEditClick}
                    >
                        <Pencil16 />
                        <FormattedMessage {...messages.commentEditMenuItem} />
                    </MenuItem>
                )}
                {canDelete && (
                    <MenuItem
                        aria-label={messages.commentDeleteMenuItem.defaultMessage}
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_DELETE}
                        data-testid="delete-comment"
                        onClick={handleDeleteClick}
                    >
                        <Trash16 />
                        <FormattedMessage {...messages.commentDeleteMenuItem} />
                    </MenuItem>
                )}
            </Media.Menu>

            {isConfirmingDelete && (
                <DeleteConfirmation
                    data-resin-component={ACTIVITY_TARGETS.COMMENT_OPTIONS}
                    isOpen={isConfirmingDelete}
                    message={messages.commentDeletePrompt}
                    onDeleteCancel={handleDeleteCancel}
                    onDeleteConfirm={handleDeleteConfirm}
                />
            )}
        </TetherComponent>
    );
};
