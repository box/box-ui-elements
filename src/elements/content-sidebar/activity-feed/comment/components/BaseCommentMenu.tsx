import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import TetherComponent from 'react-tether';
import type { ITetherConstraint } from 'tether';

import { ACTIVITY_TARGETS, ActivityTargetType } from '../../../../common/interactionTargets';
import { COMMENT_STATUS_OPEN, COMMENT_STATUS_RESOLVED } from '../../../../../constants';
import { MenuItem } from '../../../../../components/menu';
import type { FeedItemStatus } from '../../../../../common/types/feed';
import Checkmark16 from '../../../../../icon/line/Checkmark16';
import DeleteConfirmation from '../../common/delete-confirmation';
import Media from '../../../../../components/media';
import messages from '../messages';
import Pencil16 from '../../../../../icon/line/Pencil16';
import Trash16 from '../../../../../icon/line/Trash16';
import X16 from '../../../../../icon/fill/X16';

import './BaseCommentMenu.scss';

export interface BaseCommentMenuProps {
    canDelete?: boolean;
    canEdit?: boolean;
    canResolve?: boolean;
    className?: string;
    handleDeleteCancel: () => void;
    handleDeleteClick: () => void;
    handleDeleteConfirm: () => void;
    handleEditClick: () => void;
    handleMenuClose: () => void;
    handleStatusUpdate: (selectedStatus: FeedItemStatus) => void;
    isConfirmingDelete: boolean;
    isResolved: boolean;
    onSelect: (isSelected: boolean) => void;
}
export const BaseCommentMenu = ({
    canDelete,
    canEdit,
    canResolve,
    handleDeleteCancel,
    handleDeleteClick,
    handleDeleteConfirm,
    handleEditClick,
    handleMenuClose,
    handleStatusUpdate,
    isConfirmingDelete,
    isResolved,
    onSelect,
}: BaseCommentMenuProps) => {
    return (
        <TetherComponent
            attachment="top right"
            className="bcs-Comment-deleteConfirmationModal"
            constraints={[{ to: 'scrollParent', attachment: 'together' }] as ITetherConstraint[]}
            targetAttachment="bottom right"
            renderElementTag="div"
            renderElementTo="body"
        >
            <Media.Menu
                data-testid="comment-actions-menu"
                dropdownProps={{
                    onMenuOpen: () => onSelect(true),
                    onMenuClose: handleMenuClose,
                    constrainToScrollParent: true,
                }}
                className="BaseCommentMenu"
                isDisabled={isConfirmingDelete}
                menuProps={{
                    'data-resin-component': ACTIVITY_TARGETS.COMMENT_OPTIONS as ActivityTargetType,
                }}
            >
                {canResolve && isResolved && (
                    <MenuItem
                        className="bcs-Comment-unresolveComment"
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="unresolve-comment"
                        onClick={() => handleStatusUpdate(COMMENT_STATUS_OPEN)}
                    >
                        <X16 />
                        <FormattedMessage {...messages.commentUnresolveMenuItem} />
                    </MenuItem>
                )}
                {canResolve && !isResolved && (
                    <MenuItem
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="resolve-comment"
                        onClick={() => handleStatusUpdate(COMMENT_STATUS_RESOLVED)}
                    >
                        <Checkmark16 />
                        <FormattedMessage {...messages.commentResolveMenuItem} />
                    </MenuItem>
                )}
                {canEdit && (
                    <MenuItem
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_EDIT}
                        data-testid="edit-comment"
                        onClick={handleEditClick}
                    >
                        <Pencil16 />
                        <FormattedMessage {...messages.commentEditMenuItem} />
                    </MenuItem>
                )}
                {canDelete && (
                    <MenuItem
                        aria-label={messages.commentDeleteMenuItem.defaultMessage}
                        data-resin-target={ACTIVITY_TARGETS.COMMENT_OPTIONS_DELETE}
                        data-testid="delete-comment"
                        onClick={handleDeleteClick}
                    >
                        <Trash16 />
                        <FormattedMessage {...messages.commentDeleteMenuItem} />
                    </MenuItem>
                )}
            </Media.Menu>

            {isConfirmingDelete && (
                <DeleteConfirmation
                    data-resin-component={ACTIVITY_TARGETS.COMMENT_OPTIONS}
                    isOpen={isConfirmingDelete}
                    message={messages.commentDeletePrompt}
                    onDeleteCancel={handleDeleteCancel}
                    onDeleteConfirm={handleDeleteConfirm}
                />
            )}
        </TetherComponent>
    );
};
