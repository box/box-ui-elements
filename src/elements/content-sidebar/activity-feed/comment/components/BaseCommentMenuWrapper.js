// @flow
import React from 'react';
import type { BoxCommentPermission, FeedItemStatus } from '../../../../../common/types/feed';
import type { OnAnnotationEdit, OnCommentEdit } from '../types';

import { BaseCommentMenu } from './BaseCommentMenu';

export interface BaseCommentMenuWrapperProps {
    canDelete: boolean;
    canEdit: boolean;
    canResolve: boolean;
    id: string;
    isEditing: boolean;
    isInputOpen: boolean;
    isResolved: boolean;
    onAnnotationEdit?: OnAnnotationEdit | typeof undefined;
    onCommentEdit: OnCommentEdit;
    onDelete: ({ id: string, permissions?: BoxCommentPermission }) => any;
    onSelect: (isSelected: boolean) => void;
    permissions: BoxCommentPermission;
    setIsEditing: ((boolean => boolean) | boolean) => void;
    setIsInputOpen: ((boolean => boolean) | boolean) => void;
}

export const BaseCommentMenuWrapper = ({
    canDelete,
    canEdit,
    canResolve,
    id,
    isEditing,
    isInputOpen,
    isResolved,
    onAnnotationEdit,
    onCommentEdit,
    onDelete,
    onSelect,
    permissions,
    setIsEditing,
    setIsInputOpen,
}: BaseCommentMenuWrapperProps) => {
    const [isConfirmingDelete, setIsConfirmingDelete] = React.useState<boolean>(false);

    const handleDeleteConfirm = (): void => {
        onDelete({ id, permissions });
        onSelect(false);
    };

    const handleDeleteCancel = (): void => {
        setIsConfirmingDelete(false);
        onSelect(false);
    };

    const handleDeleteClick = (): void => {
        setIsConfirmingDelete(true);
        onSelect(true);
    };

    const handleEditClick = (): void => {
        setIsEditing(true);
        setIsInputOpen(true);
        onSelect(true);
    };

    const handleMenuClose = (): void => {
        if (isConfirmingDelete || isEditing || isInputOpen) {
            return;
        }
        onSelect(false);
    };

    const handleStatusUpdate = (selectedStatus: FeedItemStatus): void => {
        if (onAnnotationEdit) {
            onAnnotationEdit({ id, permissions });
        } else if (onCommentEdit) {
            onCommentEdit({ id, status: selectedStatus, hasMention: false, permissions });
        }
    };

    return (
        <BaseCommentMenu
            canDelete={canDelete}
            canEdit={canEdit}
            canResolve={canResolve}
            handleDeleteCancel={handleDeleteCancel}
            handleDeleteClick={handleDeleteClick}
            handleDeleteConfirm={handleDeleteConfirm}
            handleEditClick={handleEditClick}
            handleMenuClose={handleMenuClose}
            handleStatusUpdate={handleStatusUpdate}
            isConfirmingDelete={isConfirmingDelete}
            isResolved={isResolved}
            onSelect={onSelect}
        />
    );
};
