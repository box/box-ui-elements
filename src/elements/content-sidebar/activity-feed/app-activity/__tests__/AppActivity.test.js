import * as React from 'react';
import { shallow } from 'enzyme';

import AppActivity from '../AppActivity';
import Media from '../../../../../components/media';
import { Link } from '../../../../../components/link';

describe('elements/content-sidebar/ActivityFeed/app-activity/AppActivity', () => {
    const fakeActivityTemplate = {
        id: 'template_12345',
    };
    const fakeApp = {
        id: 'app_12345',
        icon_url: 'foo/bar/baz.jpg',
        name: 'My Application',
    };
    const fakeUser = {
        id: 'user_1',
    };
    const fakeAppActivity = {
        activity_template: fakeActivityTemplate,
        app: fakeApp,
        created_at: '2019-03-07T20:12:49.223Z',
        created_by: fakeUser,
        id: 'activity_12345',
        rendered_text: 'You did something from <a data-resin-target="foo" data-resin-action="bar" >This App</a>',
    };
    const render = (props = {}) =>
        shallow(<AppActivity isPending={false} onDelete={jest.fn()} {...fakeAppActivity} {...props} />).dive();

    test('should correctly render an app activity item', () => {
        const currentUser = {
            ...fakeUser,
        };

        expect(
            render({
                currentUser,
                permissions: {
                    can_delete: true,
                },
            }),
        ).toMatchSnapshot();
    });

    test('should render as pending if isPending flag passed in', () => {
        const currentUser = {
            ...fakeUser,
        };
        const wrapper = render({
            currentUser,
            isPending: true,
            permissions: {
                can_delete: true,
            },
        });

        expect(wrapper.find(Media).hasClass('bcs-is-pending')).toBe(true);
    });

    test('should render as pending if an error occurred', () => {
        const currentUser = {
            ...fakeUser,
        };
        const wrapper = render({
            currentUser,
            error: {},
            permissions: {
                can_delete: true,
            },
        });

        expect(wrapper.find(Media).hasClass('bcs-is-pending')).toBe(true);
    });

    test('should show the overflow menu if the current user is the one who made the activity', () => {
        const wrapper = render({
            currentUser: { ...fakeUser },
        });

        expect(wrapper.exists(Media.Menu)).toBe(true);
    });

    test('should show the overflow menu if a different user, with the correct permissions', () => {
        const wrapper = render({
            currentUser: {
                id: 'someone_else',
            },
            permissions: {
                can_delete: true,
            },
        });

        expect(wrapper.exists(Media.Menu)).toBe(true);
    });

    test('should show the overflow menu if pending', () => {
        const wrapper = render({
            currentUser: {
                id: 'someone_else',
            },
            permissions: {
                can_delete: true,
            },
            isPending: true,
        });

        expect(wrapper.exists(Media.Menu)).toBe(false);
    });

    test('should show the overflow menu if missing permissions and a different user', () => {
        const wrapper = render({
            currentUser: {
                id: 'someone_else',
            },
            permissions: {
                can_delete: false,
            },
        });

        expect(wrapper.exists(Media.Menu)).toBe(false);
    });

    test('should render app activity links and pass through any resin attributes', () => {
        const action = 'my_action';
        const target = 'my_target';
        const wrapper = render({
            currentUser: {
                id: 'someone_else',
            },
            rendered_text: `You did shared via <a data-resin-target="${target}" data-resin-action="${action}">Box</a>`,
        });
        const link = wrapper.find(Link);

        expect(link.exists()).toBe(true);
        expect(link.prop('data-resin-action')).toEqual(action);
        expect(link.prop('data-resin-target')).toEqual(target);
        expect(wrapper.exists('a')).toBe(false);
    });
});
