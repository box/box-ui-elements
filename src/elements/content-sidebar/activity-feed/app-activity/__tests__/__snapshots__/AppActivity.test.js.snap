// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/ActivityFeed/app-activity/AppActivity should correctly render an app activity item 1`] = `
<ForwardRef(ActivityCard)
  className="bcs-AppActivity"
  data-resin-feature="appActivityCardtemplate_12345"
  data-resin-target="loaded"
>
  <Media
    className=""
  >
    <MediaFigure>
      <img
        alt="My Application Icon"
        className="bcs-AppActivity-icon"
        src="foo/bar/baz.jpg"
      />
    </MediaFigure>
    <MediaBody
      className="bcs-AppActivity-body"
    >
      <TetherComponent
        attachment="top right"
        className="bcs-AppActivity-confirm"
        constraints={
          Array [
            Object {
              "attachment": "together",
              "to": "scrollParent",
            },
          ]
        }
        renderElementTag="div"
        renderElementTo={null}
        targetAttachment="bottom right"
      >
        <MediaMenu
          isDisabled={false}
        >
          <MenuItem
            onClick={[Function]}
          >
            <IconTrash
              color="#4e4e4e"
            />
            <FormattedMessage
              defaultMessage="Delete"
              id="be.appActivity.deleteMenuItem"
            />
          </MenuItem>
        </MediaMenu>
      </TetherComponent>
      <figcaption
        className="bcs-AppActivity-headline"
      >
        My Application
      </figcaption>
      <div>
        <ActivityTimestamp
          date={1551989569223}
        />
      </div>
      You did something from 
      <Link
        className="bcs-AppActivity-link"
        data-resin-action="bar"
        data-resin-target="foo"
        href=""
        key="app_actvity_link_"
        rel="roreferrer noopener"
        target="_blank"
      >
        This App
      </Link>
    </MediaBody>
  </Media>
</ForwardRef(ActivityCard)>
`;
